infra/modules/providers/azure/app-service/variables.tf

```diff
@@ -20,12 +20,24 @@ variable "resource_tags" {
   default     = {}
 }
 
-variable "app_service_name" {
+variable "app_service_config" {
   description = "The name of the app service to be created"
-  type        = map(string)
+  type        = map(map(string))
```

Comment:
We should enforce the schema of this map by leveraging a different format for the type. e.g. https://github.com/microsoft/cobalt/blob/master/infra/templates/az-isolated-service-single-region/variables.tf#L41

This allows terraform to perform type checking that fails invalid configurations at build time rather than runtime

Reply:
Another great call out! Learning a lot! Thanks!

Reply:
Replaced with the following type:
```hcl
variable "app_service_config" {
  description = "The name of the app service to be created"
  type = map(object({
    image        = string
    ad_client_id = string
  }))
  default     = {}
}
```